src/Uno.UI/UI/Xaml/UIElement.cs

```diff
@@ -548,12 +495,75 @@ internal static Matrix3x2 GetTransform(UIElement from, UIElement to)
 			return matrix;
 		}
 
+		/// <summary>
+		/// Applies to the given matrix the transformation needed to convert from parent to local element coordinates space.
+		/// </summary>
+		/// <param name="matrix">The matrix into which the layout constraints should be written</param>
+		[MethodImpl(MethodImplOptions.AggressiveInlining)]
+		internal void ApplyLayoutTransform(ref Matrix3x2 matrix)
+		{
+			// This is equivalent to matrix * Matrix3x2.CreateTranslation(layout.X, layout.Y);
+			var layoutSlot = LayoutSlotWithMarginsAndAlignments;
+			matrix.M31 += (float)layoutSlot.X;
+			matrix.M32 += (float)layoutSlot.Y;
+		}
+
+		/// <summary>
+		/// Applies to the given matrix the <see cref="RenderTransform"/>.
+		/// </summary>
+		/// <param name="matrix">The matrix into which the render transform should be written</param>
+		/// <param name="ignoreOrigin">Indicates if the <see cref="RenderTransformOrigin"/> should be ignored or not.</param>
+		[MethodImpl(MethodImplOptions.AggressiveInlining)]
+		internal void ApplyRenderTransform(ref Matrix3x2 matrix, bool ignoreOrigin = false)
+		{
+			if (RenderTransform is { } transform)
+			{
+				var transformMatrix = transform.MatrixCore;
+				if (!ignoreOrigin)
+				{
+					transformMatrix = transformMatrix.CenterOn(RenderTransformOrigin, LayoutSlotWithMarginsAndAlignments.Size);
+				}
+
+				matrix *= transformMatrix;
+			}
+		}
+
+		/// <summary>
+		/// Applies to the given matrix the constrains specific to the current element (like ScrollOffsets).
+		/// </summary>
+		/// <param name="matrix">The matrix into which the transformations should be written</param>
+		[MethodImpl(MethodImplOptions.AggressiveInlining)]
+		internal void ApplyElementCustomTransform(ref Matrix3x2 matrix)
+		{
+#if !__MACOS__ // On macOS the SCP is using RenderTransforms for scrolling and zooming which has already been included.
+			if (this is ScrollViewer sv)
+			{
+				// Scroll offsets are handled at SCP level using the IsScrollPort
+				// TODO: ZoomFactor should also be handled at the SCP level!
+
+				var zoom = sv.ZoomFactor;
+				if (zoom != 1)
+				{
+					matrix *= Matrix3x2.CreateScale(zoom);
+				}
+			}
+
+			if (IsScrollPort) // Managed SCP or custom scroller
+			{
+				matrix.M31 -= (float)ScrollOffsets.X;
+				matrix.M32 -= (float)ScrollOffsets.Y;
+			}
+#endif
+		}
+
+
+
 #if !__IOS__ && !__ANDROID__ && !__MACOS__ // This is the default implementation, but it can be customized per platform
 		/// <summary>
 		/// Note: Offsets are only an approximation which does not take in consideration possible transformations
 		///	applied by a 'UIView' between this element and its parent UIElement.
 		/// </summary>
```

Comment:
```suggestion
#if !__IOS__ && !__ANDROID__ && !__MACOS__ // This is the default implementation, but it can be customized per platform
		/// <summary>
		/// Note: Offsets are only an approximation that does not take into consideration possible transformations
		///	applied by a 'UIView' between this element and its parent UIElement.
		/// </summary>
```